### PR TITLE
make: use more portable syntax for STDERR redirect to keep OSX happy

### PR DESCRIPTION
**Pull request by ZackButcher**
fixes https://github.com/istio/istio/issues/4058

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -415,7 +415,7 @@ test: | $(JUNIT_REPORT)
 	mkdir -p $(dir $(JUNIT_UNIT_TEST_XML))
 	set -o pipefail; \
 	$(MAKE) --keep-going pilot-test mixer-test security-test broker-test galley-test common-test \
-	|& tee >($(JUNIT_REPORT) > $(JUNIT_UNIT_TEST_XML))
+	2>&1 tee >($(JUNIT_REPORT) > $(JUNIT_UNIT_TEST_XML))
```
Comment (chxchx) [inline Makefile]: You still need the pipe tho
`2>&1 | tee >($(JUNIT_REPORT) > $(JUNIT_UNIT_TEST_XML))`

**Comment by ZackButcher** [comment]
I'd really like to see at least the unit tests pass on prow...

**Comment by ZackButcher** [comment]
(waiting to see which prow things fail before submitting)

**Pull request merged**